src/java/org/apache/cassandra/config/SmallestDurationMilliseconds.java

```diff
@@ -54,4 +58,27 @@ public static SmallestDurationMilliseconds inMilliseconds(long milliseconds)
     {
         return new SmallestDurationMilliseconds(milliseconds, TimeUnit.MILLISECONDS);
     }
+
+    /**
+     * Creates a {@code SmallestDurationMilliseconds} of the specified amount of milliseconds. Custom method for special cases.
+     *
+     * @param value may be a simple number (assumed to be in milliseconds) or representation the constructor can handle
+     * @return a duration
+     */
+    public static SmallestDurationMilliseconds inMillisecondsString(String value)
+    {
+        //parse the string field value
+        Matcher matcher = VALUES_PATTERN.matcher(value);
+
+        long millis;
+        //if the provided string value is just a number, then we create a Duration Spec
+        if (matcher.matches())
+        {
+            millis = Long.parseLong(value);
```

Comment:
Apache commons has utility function, no need for regex here.
```suggestion
        if (StringUtils.isNumeric(value))
        {
            long millis = Long.parseLong(value);
```

Reply:
Thanks for the find!